### ISSUE TITLE
API-30: Support for Oracle UCP

### ISSUE THREAD

**Issue by alexsm82** [OpenLiberty/open-liberty#5867]
Add support for using Oracle UCP with Open Liberty.

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by alexsm82** [comment]
@donbourne 

1.  Yes the WAD identified likely problems areas and how they would be addressed, and those have all been addressed in the implementation.
2. I demo'd the use cases most likely to cause problems - config issues in this case, and how we make the customer aware of those issues.  I also discussed the trace points added and that we have documented how a customer can enable Oracle UCP trace.  I presented to the rest of my development team, who were not involved in the development of this feature. 

3. Kevin Lefebvre is doing SVT for this feature and will respond.

4. This is part of the product's JDBC functionality and so will be supported like the rest of JDBC: L3 - 
WAS L3: JDBC, L2 - WAS L2: CET

**Comment by dave-waddling** [comment]
Hi @alexsm82, thanks for completing the Feature Test Summary. Everything looks good to me, granting FAT Focal Approval.

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same STE Archive BOX folder.  Thanks!

**Comment by marikaj123** [comment]
@alexsm82 - Does the feature need translation, TVT/Globalization?

**Comment by alexsm82** [comment]
There were a couple of messages added, which have already been translated.  TVT/Globalization is not needed.

**Comment by marikaj123** [comment]
@alexsm82 Are the translated message in the Liberty Release build?

**Comment by alexsm82** [comment]
@marikaj123 Not quite yet, they were included in 19004 shipment 1, which was merged into integration earlier today.

**Comment by marikaj123** [comment]
I am approving since the returned translated files are in the Liberty build.

**Comment by kmlefebv** [comment]
I'm approving for SVT and agree with the serviceability comment.
